examples/example.c

```diff
@@ -52,11 +51,22 @@ void masterdump( )
 	int i;
 	printf( "==MASTER DUMP==\n" );
 
-	printf( "Received data:\n" );
-	for ( i = 0; i < mstatus.dataLength; i++ )
+	printf( "Received data: slave: %d, addr: %d, count: %d, type: %d\n",
+		mstatus.data.address, mstatus.data.index, mstatus.data.count, mstatus.data.type );
+	printf( "               values:" );
+	switch ( mstatus.data.type )
 	{
-		printf( "\t slave: %d, addr: %d, type: %d, value: %d\n",
-			mstatus.data[i].address, mstatus.data[i].reg, mstatus.data[i].dataType, mstatus.data[i].value );
+		case MODBUS_HOLDING_REGISTER:
+		case MODBUS_INPUT_REGISTER:
+			for ( i = 0; i < mstatus.data.count; i++ )
+				printf( " %d", mstatus.data.regs[i] );
+			break;
+
+		case MODBUS_COIL:
+		case MODBUS_DISCRETE_INPUT:
+			for ( i = 0; i < mstatus.data.count; i++ )
+				printf( " %d", mstatus.data.coils[i] );
```

Comment:
@Milack27  This may actually be misleading (and undocumented back then), but in `mstatus.data.coils` each one bit corresponds to a single coil value, what forces use of `modbusMaskRead` routine. Example fixed in: c0c529e

Sorry for not notifying you at review stage!